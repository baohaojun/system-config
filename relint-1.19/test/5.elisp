;;; Relint test file 5          -*- emacs-lisp -*-

;; Test let bindings
(defun test-let-inside (x y)
  (looking-at (let ((x "a")
                    (y "b"))
                (let* ((y "^")
                       (z (concat x y)))
                  z))))

(defun test-let-outside (x y)
  (let ((x "A")
        (y "B"))
    (let* ((y "^")
           (z (concat x y)))
      (looking-at z))))

;; Test setq
(defun test-setq-inside (x)
  (looking-at
   (progn
     (let ((y "A")
           (z "B"))
       (setq z "A")
       (concat "[" y z "]")))))

(defun test-setq-outside (x c)
  (setq x "[")
  (let ((y "B")
        (z "M"))
    (setq z "B")
    (looking-at (concat x y z "]"))))

(defun test-push (x)
  (let ((x (list "a")))
    (push "+" x)
    (looking-at (string-join x))))

(defun test-pop (x)
  (let ((x (list "a" "b" "^")))
    (pop x)
    (looking-at (string-join x))))

(defun test-setq-defun (x)
  (setq x "[CC]")
  (looking-at x))

(defun test-setq-lambda ()
  (lambda (y)
    (setq y "[DD]")
    (looking-at y)))

(defun f1 (x)
  (let ((y "D"))
    (setq x "E" y "E")
    (concat x y)))

(defun test-setq-inside-fun ()
  (looking-at (concat "[" (f1 "C") "]")))

(defun test-push-inside ()
  (looking-at (let ((x (list "b")))
                (push "*" x)
                (string-join x))))

(defun test-pop-inside ()
  (looking-at (let* ((x (list "u" "+" "v"))
                     (y (pop x)))
                (string-join (append x (list y))))))

(defun test-prog1 ()
  (looking-at (prog1 "[UU]" "a" "b" "c")))

(defun test-prog2 ()
  (looking-at (prog2 "a" "[VV]" "b" "c")))
