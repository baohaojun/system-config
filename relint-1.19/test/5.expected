5.elisp:5:15: In call to looking-at: Unescaped literal `^' (pos 1)
  "a^"
   .^
5.elisp:16:19: In call to looking-at: Unescaped literal `^' (pos 1)
  "A^"
   .^
5.elisp:21:4: In call to looking-at: Duplicated `A' inside character alternative (pos 2)
  "[AA]"
   ..^
5.elisp:32:17: In call to looking-at: Duplicated `B' inside character alternative (pos 2)
  "[BB]"
   ..^
5.elisp:37:17: In call to looking-at: Unescaped literal `+' (pos 0)
  "+a"
   ^
5.elisp:42:17: In call to looking-at: Unescaped literal `^' (pos 1)
  "b^"
   .^
5.elisp:46:15: In call to looking-at: Duplicated `C' inside character alternative (pos 2)
  "[CC]"
   ..^
5.elisp:51:17: In call to looking-at: Duplicated `D' inside character alternative (pos 2)
  "[DD]"
   ..^
5.elisp:59:15: In call to looking-at: Duplicated `E' inside character alternative (pos 2)
  "[EE]"
   ..^
5.elisp:62:15: In call to looking-at: Unescaped literal `*' (pos 0)
  "*b"
   ^
5.elisp:67:15: In call to looking-at: Unescaped literal `+' (pos 0)
  "+vu"
   ^
5.elisp:72:15: In call to looking-at: Duplicated `U' inside character alternative (pos 2)
  "[UU]"
   ..^
5.elisp:75:15: In call to looking-at: Duplicated `V' inside character alternative (pos 2)
  "[VV]"
   ..^
